validator.py

```diff
@@ -343,6 +354,93 @@ def validate_file_location(self, filename):
             self.logger.log(LogLevel.ERROR, "Could not open file " + filename + ". Please make sure the path is valid and the file exists.")
 
 
+    def validate_dependencies(self):
```

Comment:
Is the idea that this function will call more than just self.simple_requirements in the future? Was curious why not just call simple_requirements directly. Anyways, I'm about to pull this down and test 👍

Reply:
Yes, great question. In my future PRs you'll see that I keep adding more and more function calls to this. I wanted to keep the yaml api validation separate from the different dependency validations we are doing, which is why I've created this function instead of eventually calling a million functions from main

Reply:
Makes perfect sense, I figured there was a master plan